src/pages/volunteer page/components/Filter/Checkbox.js

```diff
@@ -0,0 +1,78 @@
+import React, { useRef, useLayoutEffect, useState } from "react";
+import { status } from "./constants";
+import { data } from "./regionsData";
+
+export default function Checkbox({
+  indeterminate,
+  checked,
+  id,
+  compute,
+  setUserChoice,
+  ...rest
+}) {
+  const [checkboxState, setCheckboxState] = useState([]);
+  const updateRegionsInState = (data, inputReference) => {
+    if (inputReference == "צפון") {
+      data[0].items.forEach((item) => {
+        setUserChoice((prev) => [...prev, item.name]);
+        // setCheckboxState((prev) => [...prev, item.name]);
+        // setUserChoice(checkboxState);
+        console.log(checkboxState);
+      });
+    }
+  };
+  const inputRef = useRef(null);
+
+  useLayoutEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.indeterminate = indeterminate;
+      inputRef.current.checked = checked;
+    }
+  }, [indeterminate, checked]);
+
+  return (
+    <input
+      {...rest}
+      ref={inputRef}
+      type="checkbox"
+      onChange={() => {
```

Comment:
usually u want to keep your jsx clean , so onChange function should only call the function that does all the logic u want